koptional/src/main/kotlin/com/gojuno/koptional/Optional.kt

```diff
@@ -16,4 +36,19 @@ object None : Optional<Nothing>() {
     override fun toString() = "None"
 }
 
-fun <T : Any> T?.toOptional(): Optional<T> = if (this == null) None else Some(this)
+/**
+ * Wrap an instance of T (or null) into an [Optional]:
+ *
+ * ```
+ * val a: String? = "str"
+ * val b: String? = null
+ *
+ * val optionalA = a.toOptional() // Some("str")
+ * val optionalB = b.toOptional() // None
+ * ```
+ *
+ * This is the preferred method of obtaining an instance of [Optional] in Kotlin. In Java, prefer
+ * using the static [Optional.toOptional] method.
+ */
+@Suppress("NOTHING_TO_INLINE")
+inline fun <T : Any> T?.toOptional(): Optional<T> = if (this == null) None else Some(this)
```

Comment:
Ah sorry for not being clear, that's not what I meant :)

Inlining this extension everywhere can inflate user's class files noticeably, we intentionally didn't make this one inline

What I was talking about is that we can copy-paste this body to the `@JvmStatic` one so we don't have to do two method calls just because we're calling it from Java

Reply:
I got your point initially, just thought inlining would be a better option here, as opposed to duplication. Since it's a one-liner I don't see it noticeably affecting the bytecode size, but I can change it to just use the same code as the other version.

Reply:
I see, well it's one-liner in Kotlin 😸 , bytecode is not that small

```
L0
    LINENUMBER 21 L0
    ALOAD 0
    IFNONNULL L1
    GETSTATIC com/gojuno/koptional/None.INSTANCE : Lcom/gojuno/koptional/None;
    CHECKCAST com/gojuno/koptional/Optional
    GOTO L2
L1
    NEW com/gojuno/koptional/Some
    DUP
    ALOAD 0
    INVOKESPECIAL com/gojuno/koptional/Some.<init> (Ljava/lang/Object;)V
    CHECKCAST com/gojuno/koptional/Optional
L2
    ARETURN
L3
    LOCALVARIABLE $receiver Ljava/lang/Object; L0 L3 0
    MAXSTACK = 3
    MAXLOCALS = 1
```

Wonder why it does CHECKCAST com/gojuno/koptional/Optional, seems useless, maybe we need to update compiler

Reply:
Yeah well, hidden costs of Kotlin!